From 1b81b495c361b5673dd72e705f16b79e8aa4798a Mon Sep 17 00:00:00 2001
From: Marco Carletti <mar.carletti@gmail.com>
Date: Fri, 8 Aug 2025 16:10:54 +0200
Subject: [PATCH] CSB-7516: relies on HTTP_URI if HTTP_PATH is empty

---
 .../rest/openapi/RestOpenApiProcessor.java      | 17 ++++++++++++++---
 1 file changed, 14 insertions(+), 3 deletions(-)

diff --git a/components/camel-rest-openapi/src/main/java/org/apache/camel/component/rest/openapi/RestOpenApiProcessor.java b/components/camel-rest-openapi/src/main/java/org/apache/camel/component/rest/openapi/RestOpenApiProcessor.java
index 626372536f7b..fa83828109a0 100644
--- a/components/camel-rest-openapi/src/main/java/org/apache/camel/component/rest/openapi/RestOpenApiProcessor.java
+++ b/components/camel-rest-openapi/src/main/java/org/apache/camel/component/rest/openapi/RestOpenApiProcessor.java
@@ -33,6 +33,7 @@ import org.apache.camel.support.processor.RestBindingAdvice;
 import org.apache.camel.support.processor.RestBindingAdviceFactory;
 import org.apache.camel.support.processor.RestBindingConfiguration;
 import org.apache.camel.support.service.ServiceHelper;
+import org.apache.commons.lang3.StringUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
@@ -94,12 +95,15 @@ public class RestOpenApiProcessor extends DelegateAsyncProcessor implements Came
     @Override
     public boolean process(Exchange exchange, AsyncCallback callback) {
         // use HTTP_URI as this works for all runtimes
-        String path = exchange.getMessage().getHeader(Exchange.HTTP_PATH, String.class);
+        String path = removeBasePath(exchange.getMessage().getHeader(Exchange.HTTP_PATH, String.class));
         //        if (path != null) {
         //            path = URISupport.stripQuery(path);
         //        }
-        if (path != null && path.startsWith(basePath)) {
-            path = path.substring(basePath.length());
+
+        //CSB-7516
+        if (StringUtils.isBlank(path)
+                && StringUtils.isNotBlank(exchange.getMessage().getHeader(Exchange.HTTP_URI, String.class))) {
+            path = removeBasePath(exchange.getMessage().getHeader(Exchange.HTTP_URI, String.class));
         }
         String verb = exchange.getMessage().getHeader(Exchange.HTTP_METHOD, String.class);
 
@@ -144,6 +148,13 @@ public class RestOpenApiProcessor extends DelegateAsyncProcessor implements Came
         return true;
     }
 
+    private String removeBasePath(String path) {
+        if (path != null && path.startsWith(basePath)) {
+            path = path.substring(basePath.length());
+        }
+        return path;
+    }
+
     @Override
     protected void doInit() throws Exception {
         super.doInit();
-- 
2.48.1

